thorlcr/master/thmastermain.cpp

```diff
@@ -650,14 +646,22 @@ int main( int argc, const char *argv[]  )
         Owned<IGroup> serverGroup = createIGroup(daliServer.str(), DALI_SERVER_PORT);
 
         unsigned retry = 0;
-        for (;;) {
-            try {
-                unsigned port = getFixedPort(TPORT_mp);
-                LOG(MCdebugProgress, thorJob, "calling initClientProcess Port %d", port);
-                initClientProcess(serverGroup, DCR_ThorMaster, port);
+        for (;;)
+        {
+            try
+            {
+                LOG(MCdebugProgress, thorJob, "calling initClientProcess %d", thorEp.port);
+                initClientProcess(serverGroup, DCR_ThorMaster, thorEp.port);
+                if (0 == thorEp.port)
```

Comment:
Not sure thorEp.port can be zero as that is handled above (line 615?)

Reply:
It can be zero if specifically set to 0 (meaning dynamic).
Which is what the helm charts do if thormaster is running in containerPerAgent: false,
which means there could be multiple thormasters in the same container and hence dynamic ports (in same way eclagent does) is what is wanted.